src/main.rs

```diff
@@ -484,6 +485,13 @@ impl<'a> System<'a> for VictoryDetector {
     }
 }
 
+fn adjust_viewport_to_window_size(gfx: std::cell::Ref<Graphics>, world: &mut World, window: &Window) {
```

Comment:
This feels a bit strange, both as it is free-standing function and with the order of arguments. Also, while not necessarily bad, I feel strange about passing the `Ref` around.

Options to make it IMO „nicer“:
* Make it a method on viewport. It would also mean pulling the `World` argument out and getting viewport out of it outside of the function, because honestly, the function doesn't need the whole world. And I think this can be done on the viewport *before* it is put into the world.
* Pass only &Graphics. No need for fancy `Ref`s and no need to `gfx.borrow()` inside (which I suspect is extraneous now too).

Reply:
I fully agree - I got my stamina drained on returning to Rust and understanding it after long pause, so I was so happy it works I forgot to do the final re-evaluation and design cleanup.

I agree, and will think how to achieve the proposed changes. Thank you!